core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -242,6 +242,42 @@ case class PSBT(
     PSBT(globalMap, newInputMaps, outputMaps)
   }
 
+  def addScriptWitnessToInput(
+      scriptWitness: ScriptWitness,
+      index: Int): PSBT = {
+    require(
+      index < inputMaps.size,
+      s"index must be less than the number of input maps present in the psbt, $index >= ${inputMaps.size}")
+    require(!inputMaps(index).isFinalized,
+            s"Cannot update an InputPSBTMap that is finalized, index: $index")
+
+    val previousElements = inputMaps(index).elements
+      .filterNot(_.key.head == PSBTInputKeyId.WitnessScriptKeyId.byte)
+
+    val newElement = scriptWitness match {
+      case p2wpkh: P2WPKHWitnessV0 =>
+        InputPSBTRecord.WitnessScript(P2PKHScriptPubKey(p2wpkh.pubKey))
+      case p2wsh: P2WSHWitnessV0 =>
+        InputPSBTRecord.WitnessScript(p2wsh.redeemScript)
+      case _ =>
+        throw new IllegalArgumentException(
+          s"Invalid scriptWitness given, got: $scriptWitness")
+    }
+
+    val compressedMap = InputPSBTMap(previousElements :+ newElement)
+      .compressMap(transaction.inputs(index))
+    val newMap = if (scriptWitness.isInstanceOf[P2WPKHWitnessV0]) {
+      // Compress to remove NonWitnessOrUnknownUTXO then remove P2WPKHWitnessV0
```

Comment:
any reason this isn't done the pattern match above? It seems like this `if(scriptWitness.isInstanceOf[P2WPKHWitnessV0])` isn't necessary.